main.py

```diff
@@ -1,11 +1,126 @@
+            # NOTE: quality parameter was chosen experimentally
+            im.save(buffer, optimize=True, quality=45, format='JPEG')
+            image = buffer.getvalue()
+            buffer.close()
+
+            # To prevent integer overflow in bytes sending
+            if current_image_number == 2 ** 32 - 1:
+                current_image_number = 0
+
+            for i in range(len(image) // CHUNK_SIZE + (len(image) / CHUNK_SIZE != 0)):
+                sock.send(b''.join([struct.pack('i', len(image)), struct.pack('i', current_image_number),
+                                    struct.pack('i', i), image[i * CHUNK_SIZE:(i + 1) * CHUNK_SIZE]]))
+
+            current_image_number += 1
+            logger.root_logger.debug(f"Camera. Got and sent {len(image)} bytes")
+            time.sleep(0)
+
+    except (BrokenPipeError, ConnectionResetError) as e:
+        logger.root_logger.warning(e)
+
+
+def receive_data(sock, queue=b'', resolution=(640, 480)):
+    sock.settimeout(100)
+    log = logging.getLogger("Camera")
+    window_display = pygame.display.set_mode(resolution)
+
+    current_image_number = 0
+    try:
+        while True:
+            if len(queue) != 0:
+                data = [queue]
+            else:
+                data = []
+
+            # To prevent integer overflow in bytes sending
+            if current_image_number == 2 ** 32 - 1:
+                current_image_number = 0
+
+            while True:
+                try:
+                    pack = sock.recv(CHUNK_SIZE + INFORMATION_SIZE)
+                    while struct.unpack('i', pack[4:8])[0] < current_image_number:
+                        pack = sock.recv(CHUNK_SIZE + INFORMATION_SIZE)
+                        pass
+                    if struct.unpack('i', pack[4:8])[0] == current_image_number:
+                        data += [pack]
+                    if struct.unpack('i', pack[4:8])[0] > current_image_number:
+                        queue = pack
+                        break
+                except TimeoutError:
+                    pass
+
+            current_image_number += 1
+
+            data = list(set(data))
+            data.sort(key=lambda data_item: struct.unpack('i', data_item[8:12])[0])
+
+            image = b''
+            error_key = 0
+            index_should_be = 0
+            for i in range(len(data)):
+                data_index = struct.unpack('i', data[i][8:12])[0]
+                if data_index - index_should_be > 0:
+                    error_key = 1
+                    break
+
+                image += data[i][12:]
+                index_should_be += 1
+
+            if struct.unpack('i', data[0][:4])[0] - len(image) > 0 or error_key:
+                log.warning("file corrupted")
+                time.sleep(0)
+                continue
+
+            try:
+                buffer = BytesIO()
+                buffer.write(image)
```

Comment:
I think you can do it like that:

```suggestion
                buffer = BytesIO(image)
```